casual/casual-caller/src/test/groovy/se/laz/casual/connection/caller/CasualCallerImplTest.groovy

```diff
@@ -286,27 +275,25 @@ class CasualCallerImplTest extends Specification
     def 'TPNOTRAN tpcall, in transaction'()
     {
        given:
-       1 * transactionManager.suspend() >> {
-          Mock(Transaction)
+       TransactionLess transactionLess = Mock(TransactionLess) {
+          0 * tpcall(_)
        }
-       def caller = new CasualCallerImpl(lookup, connectionFactoryProvider, new TransactionLess(transactionManagerProvider))
+       def caller = new CasualCallerImpl(lookup, connectionFactoryProvider, transactionLess)
        caller.tpCaller = Mock(TpCallerFailover)
-       1 * transactionManager.resume(_)
        when:
-       caller.tpcall("foo", Mock(CasualBuffer), Flag.of(AtmiFlags.TPNOTRAN))
+       caller.tpcall("foo", Mock(CasualBuffer), Flag.of(AtmiFlags.NOFLAG))
```

Comment:
Why send NOFLAG in this test and expect 0 tpcalls in transactionLess? Seems contrary to test description where I would expect TPNOTRAN flag is sent and tpCall is invoked for the call in transactionLess

Reply:
No flag then it is in transaction.
We only check if TPNOTRAN is sent to us via that API, if it is - then it should be transactionless tp(a)call.

Reply:
It was wrong before.
With CasualCaller::tpcall(AtmiFlags.NOFLAG) - it is transactional -> TransactionLess is not used.
It is only used if you provide TPNOTRAN.

Reply:
Yes, but the test probably shouldn't be called 'TPNOTRAN tpcall, in transaction' if the tpcall is done with no flags. Just 'tpcall, in transaction' maybe?